feat(allocator)!: remove `Vec::into_boxed_slice` method

Dunqing opened the pull request:
related: https://github.com/oxc-project/oxc/pull/9656#discussion_r1992578706

This method blocks us from replacing the allocator-api2's `Vec` with our own `Vec`(now called `Vec2`) in #9656, considering no use cases for this method, just remove it instead. After our own `Vec` is feature-complete, we can consider making https://github.com/oxc-project/oxc/blob/b32469a458b36097162e45c8ab20327d46867c7f/crates/oxc_allocator/src/vec2/mod.rs#L1686-L1719 this work.

Dunqing commented:
* **#9735** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9735?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9735?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9735).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

overlookmotel commented:
### Merge activity

* **Mar 13, 12:17 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
feat(allocator)!: remove `Vec::into_boxed_slice` method (#9735)

related: https://github.com/oxc-project/oxc/pull/9656#discussion_r1992578706

This method blocks us from replacing the allocator-api2's `Vec` with our own `Vec`(now called `Vec2`) in #9656, considering no use cases for this method, just remove it instead. After our own `Vec` is feature-complete, we can consider making https://github.com/oxc-project/oxc/blob/b32469a458b36097162e45c8ab20327d46867c7f/crates/oxc_allocator/src/vec2/mod.rs#L1686-L1719 this work.
